docs/reproducibility.md

```diff
@@ -8,48 +8,152 @@ The assumption here is that the original build was done using a Docker.
 
 - Decode the report and save it to a file:
 
     ```sh
-    cat node-quote.json | jq -r '.raw' | base64 -d > snp-report.bin
+    $ cat node-quotes.json | jq -r '.quotes[0].raw' | base64 -d > snp-report.bin
     ```
 
-- Inspect the report to get the measurements. You could use https://github.com/virtee/snpguest to display the report details:
+- Inspect the report to get the measurements. You could use https://github.com/virtee/snpguest to display the report details, e.g.:
 
     ```sh
-    snpguest display report snp-report.bin
+    $ curl -LO https://github.com/virtee/snpguest/archive/refs/tags/v0.8.0.tar.gz
+    $ tar -xvf v0.8.0.tar.gz
+    $ cd snpguest-0.8.0
+    $ cargo build -r
+    $ cd ..
+    $ ./snpguest-0.8.0/target/release/snpguest display report snp-report.bin
+    <...>
+    Measurement:
+    18 25 a4 bf 2a 9c 38 35 66 a7 17 63 26 83 9a c0
+    e3 6a 1c 5b 37 e9 e6 fa bc 8f dd 71 30 d5 8c ef
+    56 f4 34 75 02 b9 47 89 53 0c ec 19 8a a5 15 43
+
+    Host Data:
+    5a e7 b1 4e e0 c9 c4 fe 26 7d 19 1f 25 b2 0f ff
+    e2 4e 29 c4 ac 41 9c 50 50 1d 20 c8 69 bb ba 65
+    <...>
     ```
 
+- You could also verify the provided report with services such as Microsoft Azure Attestation Service, this step is excluded for the brevity reasons
+
+### Extract image layers from security policy
+
+Inspect the service join policy content and extract the Rego policy used to validate the container:
+
+```sh
+$ cat service-join-policy.json | jq -r '.snp.hostData["5ae7b14ee0c9c4fe267d191f25b20fffe24e29c4ac419c50501d20c869bbba65"]' | printf "%s" "$(cat)" > ccepolicy.rego
+```
+
+_`printf "%s" "$(cat)"` is important to not to create additional line in the file as otherwise sha256 digests will not match._
+
+The policy is used by the Utility VM (UVM) to launch a container group (this ledger application). It contains the image layers we want to compare and be able to reproduce.
+
+You can use an OPA agent to convert the Rego file to JSON to then select layers or do that manually, e.g.:
+
+```sh
+$ curl -s -LO https://openpolicyagent.org/downloads/v1.1.0/opa_linux_amd64_static
+$ chmod 755 opa_linux_amd64_static
+$ ./opa_linux_amd64_static parse ccepolicy.rego -f json > ccepolicy.json
+
+# jq find the containers rule and extract layers of each container
+$ cat ccepolicy.json | jq '[ .rules[] | select(.head.name == "containers") | .head.value.value[].value[] | select(.[].value == "layers") | .[1].value | map(.value) ]' > containerlayers.json
```

Comment:
I believe the field where the image layers are specified in not the same across different SNP platforms (C-ACI and C-AKS). I suggest specifying this only applies to C-ACI security policies generated using the `az confcom acipolicygen` CLI.

Reply:
added a note